src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -565,6 +568,33 @@ private static final void addConnectorOptions(ConfigDef config) {
     ).define(
         TOPIC_PREFIX_CONFIG,
         Type.STRING,
+        "",
+        new Validator() {
+          @Override
+          public void ensureValid(final String name, final Object value) {
+            if (value == null) {
+              throw new ConfigException(name, value, "Topic prefix must not be null.");
+            }
+
+            if (!((String) value).isEmpty()) {
+              char first = ((String) value).charAt(0);
+              if (Stream.of('_', '-', '.').collect(Collectors.toSet()).contains(first)) {
+                throw new ConfigException(name, value,
+                    "Topic prefix must not start with '_', '-' or '.'.");
+              }
+            }
+
+            if (((String) value).length() > 249) {
+              throw new ConfigException(name, value,
+                  "Topic prefix length must not exceed max topic name length, 249 chars");
+            }
+
+            if (((String) value).trim().contains(" ")) {
```

Comment:
Shouldn't we check for all of the other possible invalid characters here as well?